aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -81,6 +94,12 @@ private GetPolicyResponse getExistingPolicy(AmazonWebServicesClientProxy proxy,
         );
     }
 
+    // Since we don't have a mechanism of determining the changeset, we perform a delete and then create for the updated channelUris
+    private void updateNotificationChannels(String pgName, AmazonWebServicesClientProxy proxy, List<String> channelUris) {
+        deleteNotificationChannels(pgName, proxy, profilerClient);
+        addChannelNotifications(pgName, channelUris, proxy, profilerClient);
+    }
```

Comment:
This is a really bad approach, because it means that customers can miss notifications between delete and add. Furthermore, if the add then fails, we'll leave the profiling group in a state where there are NO notifications, rather than in its previous, valid state.

For that reason, what we must do here is retrieve the current notification channels, and then determine our own "changeset". It's still not perfect, because someone can call the API while we're doing this, but at least we guarantee that we never remove something we shouldn't, which is where we want to be.